python/cuml/common/sparsefuncs.py

```diff
@@ -208,14 +208,16 @@ def extract_knn_graph(knn_graph, convert_dtype=True):
         knn_indices = knn_graph.col
 
     if knn_indices is not None:
+        convert_to_dtype = None
+        if convert_dtype:
+            convert_to_dtype = np.int32 if sparse else np.int64
```

Comment:
It's going to be important to change this when FAISS is updated (and the indices are 32-bit). Referencing relevant issue: https://github.com/rapidsai/cuml/issues/2821